bundle/config/mutator/resolve_variable_references.go

```diff
@@ -205,3 +227,45 @@ func (m *resolveVariableReferences) resolveOnce(b *bundle.Bundle, prefixes []dyn
 
 	return hasUpdates, diags
 }
+
+// selectivelyMutate applies a function to a subset of the configuration
+func (m *resolveVariableReferences) selectivelyMutate(b *bundle.Bundle, fn func(value dyn.Value) (dyn.Value, error)) error {
+	included := []string{
+		"variables",
+		"bundle",
+		"include",
+		"workspace",
+		"artifacts",
+		"targets",
+		"environments",
+		"sync",
+		"run_as",
+		"presets",
+		"experimental",
+		"permissions",
+	}
+
+	if m.includeResources {
+		included = append(included, "resources")
```

Comment:
The intention is to process resources only, right? Why do we construct a slice with all keys then?

Reply:
Because we need the rest of the bundle to get variable values from it